pkg/logs/tailers/file/tailer_windows_test.go

```diff
@@ -0,0 +1,182 @@
+	"github.com/DataDog/datadog-agent/pkg/logs/internal/decoder"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/logs/metrics"
+	"github.com/DataDog/datadog-agent/pkg/logs/sources"
+	status "github.com/DataDog/datadog-agent/pkg/logs/status/utils"
+	"github.com/DataDog/datadog-agent/pkg/logs/types"
+	"github.com/DataDog/datadog-agent/pkg/logs/util/opener"
+)
+
+// newTestTailer creates a tailer configured similarly to suite setup in tailer_test.go
+// but without starting readForever; it allows direct calls to readAvailable.
+func newTestTailer(t *testing.T, testPath string, fingerprint *types.Fingerprint, fingerprinter Fingerprinter, opener opener.FileOpener, decoderOptions *decoder.MockDecoderOptions) (*Tailer, chan *message.Message) {
+	t.Helper()
+
+	outputChan := make(chan *message.Message, 20)
+	source := sources.NewReplaceableSource(sources.NewLogSource("", &config.LogsConfig{
+		Type: config.FileType,
+		Path: testPath,
+	}))
+	sleepDuration := 1 * time.Millisecond
+	info := status.NewInfoRegistry()
+
+	decoder := decoder.NewMockDecoderWithOptions(decoderOptions)
+
+	tailerOptions := &TailerOptions{
+		OutputChan:      outputChan,
+		Decoder:         decoder,
+		File:            NewFile(testPath, source.UnderlyingSource(), false),
+		SleepDuration:   sleepDuration,
+		Info:            info,
+		CapacityMonitor: metrics.NewNoopPipelineMonitor("").GetCapacityMonitor("", ""),
+		Registry:        auditor.NewMockRegistry(),
+		Fingerprint:     fingerprint,
+		Fingerprinter:   fingerprinter,
+		FileOpener:      opener,
+	}
+
+	tailer := NewTailer(tailerOptions)
+	tailer.setup(0, io.SeekStart)
+
+	return tailer, outputChan
+}
+
+func TestReadAvailable(t *testing.T) {
+	mockFile := opener.NewMockFile("test.log", [][]byte{
+		[]byte("one\ntwo\n"),
+		[]byte("three\nfour\n"),
+	})
+
+	opener := opener.NewMockFileOpener()
+	opener.AddMockFile(mockFile)
+
+	fingerprinterMock := NewFingerprinterMock()
+	tailer, _ := newTestTailer(t, mockFile.Name(), nil, fingerprinterMock, opener, nil)
+	n, err := tailer.readAvailable()
+	assert.ErrorIs(t, err, io.EOF, "Expected EOF, got %v", err)
+	assert.Equal(t, mockFile.FileSize(), n, "Expected to have read the entire file of %d bytes, got %d", mockFile.FileSize(), n)
+
+	assert.Equal(t, 2, len(tailer.decoder.InputChan()), "Expected 2 messages to have been decoded")
+}
+
+func TestReadAvailableRotation(t *testing.T) {
+	mockFile := opener.NewMockFile(
+		"test.log",
+		[][]byte{
+			[]byte("one\ntwo\n"),
+			[]byte("three\nfour\n"),
+		},
+		[][]byte{
+			[]byte("five\nsix\n"),
+		},
+	)
+
+	firstFileSize := mockFile.FileSize()
+
+	opener := opener.NewMockFileOpener()
+	opener.AddMockFile(mockFile)
+
+	mockDecoderOptions := &decoder.MockDecoderOptions{
+		InputChanSize:  0,
+		OutputChanSize: 0,
+	}
+
+	tailer, _ := newTestTailer(t, mockFile.Name(), nil, nil, opener, mockDecoderOptions)
+	tailer.windowsOpenFileTimeout = 0
+	go func() {
+		time.Sleep(10 * time.Millisecond)
```

Comment:
NIT: Could we add a comment to this goroutine? 
Reading the code - it's not immediately clear what this does. 
Questions that come to mind:
- why do we need to sleep before a blocking channel read?
- what are we draining from the decoder input before starting the decoder? 
(I'm sure I could figure this out with a more careful examination of the code, but a comment would be helpful here)